### ISSUE TITLE
App crashes during initialization on Samsung Galaxy Tab S4

### ISSUE THREAD

**Issue by dev4711** [google-ar/arcore-android-sdk#671]
### SPECIFIC ISSUE ENCOUNTERED
The app crashes during the initialization process of ArCore or Sceneform.

### VERSIONS USED
- Android Studio: 3.2.1
- ARCore SDK for Android: 1.6
- Device manufacturer, model, and O/S: Samsung Galaxy Tab S4 (SM-T830) with Android 8.1.0
- ARCore: 1.6.181112096
- Output of `adb shell getprop ro.build.fingerprint`: samsung/gts4lwifixx/gts4lwifi:8.1.0/M1AJQ/T830XXU2ARJ3:user/release-keys

### COMMENTS
* Seams to be a similar crash like #542 
* I'm not sure where to put this issue. The crash itself is in ArCore but I use Sceneform which is also in the stacktrace of the crash. If it doesn't fit feel free to move it.

### STACKTRACE
```
2018-12-10 09:52:50.684 9890-10565/? E/ACameraDevice: onDeviceIdle sending state cb
2018-12-10 09:52:50.703 716-11143/? E/mm-camera: <IFACE ><ERROR> 596: iface_ispif_proc_close: 144 closed
2018-12-10 09:52:50.703 716-11143/? E/mm-camera: <IFACE ><ERROR> 604: iface_ispif_proc_close: ispif->open_count 0
2018-12-10 09:52:50.706 716-11143/? E/mm-camera: <SENSOR><ERROR> 258: module_sensor_companion_stream_off: This does not have CC module.
2018-12-10 09:52:50.711 716-11143/? E/mm-camera: <MCT   ><ERROR> 193: stop_sof_check_thread: Returning as SOF timer thread not yet initialized
2018-12-10 09:52:50.712 716-11143/? E/mm-camera: <MCT   ><ERROR> 376: mct_stream_link_module_array: stream_type_mod[0][0] name : (null)
2018-12-10 09:52:50.715 1001-5172/? E/CameraDeviceClient: Disconnect from CameraDeviceClient
2018-12-10 09:52:50.716 1001-5172/? E/Camera2-FrameProcessorBase: removeListener: 
2018-12-10 09:52:50.726 716-1681/? E/QCamera: <HAL><ERROR> stop: 1399: Attempt to stop inactive channel
2018-12-10 09:52:50.726 716-1681/? E/QCamera: <HAL><ERROR> stop: 261: Attempt to stop inactive channel
2018-12-10 09:52:50.726 716-1681/? E/QCamera: <HAL><ERROR> stop: 261: Attempt to stop inactive channel
2018-12-10 09:52:50.727 716-11143/? E/mm-camera: <MCT   ><ERROR> 193: stop_sof_check_thread: Returning as SOF timer thread not yet initialized
2018-12-10 09:52:51.003 716-1681/? E/mm-camera: <MCT   ><ERROR> 5597: mct_pipeline_stop_session: Stop session successful for session 4
2018-12-10 09:52:51.003 716-1681/? E/mm-camera: <MCT   ><ERROR> 193: stop_sof_check_thread: Returning as SOF timer thread not yet initialized
2018-12-10 09:52:51.627 1477-1642/? E/WindowManager: win=Window{5c4da41 u0 com.example.application/com.example.application.activity.ArSessionActivity EXITING} destroySurfaces: appStopped=false win.mWindowRemovalAllowed=false win.mRemoveOnExit=false win.mViewVisibility=8, caller=com.android.server.wm.AppWindowToken.destroySurfaces:748 com.android.server.wm.AppWindowToken.destroySurfaces:732 com.android.server.wm.WindowState.onExitAnimationDone:5523 com.android.server.wm.AppWindowAnimator.stepAnimationLocked:517 com.android.server.wm.AppWindowToken.stepAppWindowsAnimation:1745 
2018-12-10 09:52:52.017 9890-9890/? E/NdkImageReader: AImageReader_getWindow
2018-12-10 09:52:52.020 1001-5172/? E/CameraService: checkTemperature (0)
2018-12-10 09:52:52.020 1001-5172/? E/CameraService: setCompanionDisableSysFs : cc.disable property set to 0
2018-12-10 09:52:52.044 716-1681/? E/mm-camera: <SENSOR><ERROR> 2276: hwinfo_make_3a_name: 3a name (type : 0) Y13UL_libTsAe.so
2018-12-10 09:52:52.044 716-1681/? E/mm-camera: <SENSOR><ERROR> 2301: hwinfo_make_3a_bin_path: path=/data/camera/TsAeTuningData.bin
2018-12-10 09:52:52.045 716-1681/? E/mm-camera: <SENSOR><ERROR> 2276: hwinfo_make_3a_name: 3a name (type : 2) Y13UL_libTsAwb.so
2018-12-10 09:52:52.045 716-1681/? E/mm-camera: <SENSOR><ERROR> 2301: hwinfo_make_3a_bin_path: path=/data/camera/TsAwbTuningData.bin
2018-12-10 09:52:52.046 716-1681/? E/mm-camera: <SENSOR><ERROR> 2276: hwinfo_make_3a_name: 3a name (type : 3) Y13UL_libTsAccm.so
2018-12-10 09:52:52.046 716-1681/? E/mm-camera: <SENSOR><ERROR> 2301: hwinfo_make_3a_bin_path: path=/data/camera/TsAccmTuningData.bin
2018-12-10 09:52:52.046 716-1681/? E/mm-camera: <SENSOR><ERROR> 2276: hwinfo_make_3a_name: 3a name (type : 1) Y13UL_libTsAf.so
2018-12-10 09:52:52.046 716-1681/? E/mm-camera: <SENSOR><ERROR> 2301: hwinfo_make_3a_bin_path: path=/data/camera/TsAfTuningData.bin
2018-12-10 09:52:52.055 716-1681/? E/mm-camera: <STATS ><ERROR> 2996: stats_port_check_caps_reserve: Invalid Port capability type!
2018-12-10 09:52:52.055 716-1681/? E/mm-camera: <STATS ><ERROR> 2996: stats_port_check_caps_reserve: Invalid Port capability type!
2018-12-10 09:52:52.057 716-11231/? E/mm-camera: <SENSOR><ERROR> 124: module_sensor_offload_init_config: This does not have OIS module.
2018-12-10 09:52:52.057 716-11231/? E/mm-camera: <SENSOR><ERROR> 140: module_sensor_offload_init_config: This does not have CC module.
2018-12-10 09:52:52.059 1001-5172/? E/Camera2-FrameProcessorBase: FrameProcessorBase: created
2018-12-10 09:52:52.059 1001-5172/? E/Camera2-FrameProcessorBase: registerListener: Registering listener for frame id range 0 - 2147483647
2018-12-10 09:52:52.061 716-1682/? E/Camera2-Metadata: Mismatched tag type when updating entry enable (-2146762752) of type byte; got type int32 data instead 
2018-12-10 09:52:52.061 716-1682/? E/Camera2-Metadata: Mismatched tag type when updating entry is_main (-2146762751) of type byte; got type int32 data instead 
2018-12-10 09:52:52.065 716-11229/? E/mm-camera: <MCT   ><ERROR> 376: mct_stream_link_module_array: stream_type_mod[0][0] name : (null)
2018-12-10 09:52:52.130 9890-9890/? E/native: session.cc:1232 FrameHitTest while not tracking. Returning empty list.
2018-12-10 09:52:52.201 1477-1540/? E/ANDR-PERF-JNI: Iop tryGetService failed
2018-12-10 09:52:52.304 716-11231/? E/mm-camera: <SENSOR><ERROR> 457: module_sensor_offload_init_config: This does not have CC module.
2018-12-10 09:52:52.305 716-11231/? E/mm-camera: <SENSOR><ERROR> 497: module_sensor_offload_init_config: Success
2018-12-10 09:52:52.310 716-11229/? E/mm-camera: <SENSOR><ERROR> 4766: module_sensor_stream_on: ide 40002 stream type 1 This is not have CC module.
2018-12-10 09:52:52.310 716-11229/? E/mm-camera: <SENSOR><ERROR> 3097: sensor_check_stream: Sensor_StreamOff_Success Frame Counter: 0xff count 0
2018-12-10 09:52:52.311 716-11229/? E/mm-camera: <SENSOR><ERROR> 4863: module_sensor_stream_on: module_sensor_companion_stream_off
2018-12-10 09:52:52.311 716-11229/? E/mm-camera: <SENSOR><ERROR> 4385: sensor_set_shot_mode: test: 0
2018-12-10 09:52:52.312 716-11229/? E/mm-camera: <STATS_AF ><ERROR> 3333: af_port_handle_sensor_update: Depth_Err: actuator sensitivity: 0.000000 is invalid
2018-12-10 09:52:52.312 716-11239/? E/mm-3a-external: [cam_id_0][CMG_EFL] void AfLib_AlgorithmInit(uint8_t) (line 461) : Booting Init is not done!!
2018-12-10 09:52:52.313 716-11239/? E/mm-3a-external: [cam_id_0][CMG_EFL] uint8_t AfLib_BootingInit(uint8_t) (line 80) : Rear Version = 0xc00f 
2018-12-10 09:52:52.313 716-11239/? E/mm-3a-external: [CMG_AF] uint8_t AfLib_BootingInit(uint8_t) (line 86) : Chip Type = 1 (0:LSI, 1:QC) 
2018-12-10 09:52:52.313 716-11239/? E/mm-3a-external: [cam_id_0][CMG_EFL] uint8_t AfLib_BootingInit(uint8_t) (line 276) : af cal data far: 237, middle: 65535, near: 345, add_80: 0, add_50: 0, plus_offset: 0, minus_offset: 0 
2018-12-10 09:52:52.313 716-11229/? E/mm-camera: <ISP   ><ERROR> 66: gtm46_configure_from_chromatix_bit: submod_enable=0
2018-12-10 09:52:52.313 716-11229/? E/mm-camera: <ISP   ><ERROR> 1969: isp_sub_module_util_configure_from_chromatix_bit: LTM disable
2018-12-10 09:52:52.313 716-11229/? E/mm-camera: <STATS_ASD ><ERROR> 4423: parse_ASD_chromatix: Tuning Error: ASD uses only bayer stats
2018-12-10 09:52:52.313 716-11229/? E/mm-camera: <STATS ><ERROR> 4520: parse_chromatix_iot: error: chromatix_ptr is NULL
2018-12-10 09:52:52.314 716-11229/? E/mm-camera: <SENSOR><ERROR> 5844: module_sensor_set_start_companion_stream_on: E
2018-12-10 09:52:52.314 716-11229/? E/mm-camera: <SENSOR><ERROR> 233: module_sensor_companion_stream_on: This does not have CC module.
2018-12-10 09:52:52.314 716-11229/? E/mm-camera: <ISP   ><ERROR> 962: gtm46_set_hdr_mode: use_gtm_fix_curve = 1  submod_enable = 0 
2018-12-10 09:52:52.316 716-11229/? E/mm-camera: <IFACE ><ERROR> 647: iface_ispif_proc_open: ispif->open_count 1, fd 144
     
2018-12-10 09:52:52.324 716-11229/? E/mm-camera: <SENSOR><ERROR> 5790: module_sensor_set_start_stream_on: ide 40002 SENSOR_START_STREAM
2018-12-10 09:52:52.324 716-11229/? E/mm-camera: <SENSOR><ERROR> 3185: sensor_set_start_stream: Sensor stream ON for s5k3l2xx 
2018-12-10 09:52:52.325 716-11229/? E/mm-camera: <SENSOR><ERROR> 4766: module_sensor_stream_on: ide 40003 stream type 5 This is not have CC module.
2018-12-10 09:52:52.325 716-11229/? E/mm-camera: <SENSOR><ERROR> 3097: sensor_check_stream: Sensor_StreamOff_Success Frame Counter: 0xff count 0
2018-12-10 09:52:52.328 716-11229/? E/mm-camera: <SENSOR><ERROR> 4766: module_sensor_stream_on: ide 40004 stream type 11 This is not have CC module.
2018-12-10 09:52:52.328 716-11229/? E/mm-camera: <SENSOR><ERROR> 3097: sensor_check_stream: Sensor_StreamOff_Success Frame Counter: 0xff count 0
2018-12-10 09:52:52.331 716-11229/? E/mm-camera: <ISP   ><ERROR> 401: color_xform46_config: color_xform46_config: color xform: 0 burstmode =0
2018-12-10 09:52:52.331 716-11229/? E/mm-camera: <ISP   ><ERROR> 401: color_xform46_config: color_xform46_config: color xform: 0 burstmode =0
2018-12-10 09:52:52.342 716-11239/? E/mm-3a-external: [cam_id_0][CMG_EFL] void AfLib_AlgorithmInit(uint8_t) (line 467) : paf_stats_valid : 0, caf_stats_valid : 1
2018-12-10 09:52:52.342 716-11239/? E/mm-3a-external: [cam_id_0][CMG_EFL] void AfLib_AlgorithmInit(uint8_t) (line 501) mode = 3
2018-12-10 09:52:52.344 716-11239/? E/mm-3a-external: [cam_id_0][CMG_EFL] void AfLib_AlgorithmInit(uint8_t) (line 467) : paf_stats_valid : 0, caf_stats_valid : 1
2018-12-10 09:52:52.344 716-11239/? E/mm-3a-external: [cam_id_0][CMG_EFL] void AfLib_AlgorithmInit(uint8_t) (line 501) mode = 3
2018-12-10 09:52:52.344 716-11230/? E/mm-camera: <SENSOR><ERROR> 6510: module_sensor_event_control_set_parm: [CAM_INTF_PARM_OIS_MODES][sub_id FAIL]
2018-12-10 09:52:52.344 716-11230/? E/mm-camera: <SENSOR><ERROR> 156: module_sensor_companion_wdr_on_off: This does not have CC module.
2018-12-10 09:52:52.362 716-11239/? E/mm-3a-external: [cam_id_0] PreviewCafExecute. delay_count_for_roi_setting[0](1)
2018-12-10 09:52:52.395 716-11239/? E/mm-3a-external: [cam_id_0][CMG_EFL] void UpdateAfStatus(AfState) (line 269) : AF status 0x1 0x0
2018-12-10 09:52:52.409 716-11233/? E/mm-camera: <ISP   ><ERROR> 962: gtm46_set_hdr_mode: use_gtm_fix_curve = 1  submod_enable = 0 
2018-12-10 09:52:52.451 713-740/? E/ANDR-PERF-REQUEST: Failed to set timer
2018-12-10 09:52:52.451 9890-11225/? E/native: camera_image_stream.cc:107 Failed to calculate timebase offset for camera stream.
2018-12-10 09:52:52.451 9890-11225/? A/native: statusor.cc:21 Attempting to fetch value instead of handling error generic::invalid_argument: CLOCK_MONOTONIC is ahead of CLOCK_BOOTTIME by 6.042us. Max is 5us
2018-12-10 09:52:52.451 9890-11225/? A/native: terminating.
    
    
    --------- beginning of crash
2018-12-10 09:52:52.452 9890-11225/? A/libc: Fatal signal 6 (SIGABRT), code -6 in tid 11225 (ImageReader-640), pid 9890 (example.app)
2018-12-10 09:52:52.455 9890-9890/? E/native: status.cc:156 generic::invalid_argument: CLOCK_MONOTONIC is ahead of CLOCK_BOOTTIME by 6.094us. Max is 5us
2018-12-10 09:52:52.462 716-11239/? E/mm-3a-external: [cam_id_0][CMG_EFL] void UpdateAfStatus(AfState) (line 269) : AF status 0x2 0x1
2018-12-10 09:52:52.478 9890-9890/? E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.example.application, PID: 9890
    com.google.ar.core.exceptions.FatalException
        at com.google.ar.core.au.a(Session.java:2)
        at com.google.ar.core.Session.throwExceptionFromArStatus(Session.java:105)
        at com.google.ar.core.Session.nativeUpdate(Native Method)
        at com.google.ar.core.Session.update(Session.java:50)
        at com.google.ar.sceneform.ArSceneView.onBeginFrame(Unknown Source:10)
        at com.google.ar.sceneform.SceneView.doFrame(Unknown Source:16)
        at android.view.Choreographer$CallbackRecord.run(Choreographer.java:909)
        at android.view.Choreographer.doCallbacks(Choreographer.java:723)
        at android.view.Choreographer.doFrame(Choreographer.java:655)
        at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:897)
        at android.os.Handler.handleCallback(Handler.java:790)
        at android.os.Handler.dispatchMessage(Handler.java:99)
        at android.os.Looper.loop(Looper.java:164)
        at android.app.ActivityThread.main(ActivityThread.java:7002)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:441)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1408)
2018-12-10 09:52:52.495 716-11239/? E/mm-3a-external: [cam_id_0] PreviewCafExecute. delay_count_for_roi_setting[0](1)
2018-12-10 09:52:52.505 11274-11274/? E/propClient: PropClient failed to load
2018-12-10 09:52:52.559 11276-11276/? A/DEBUG: *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***
2018-12-10 09:52:52.560 11276-11276/? A/DEBUG: Build fingerprint: 'samsung/gts4lwifixx/gts4lwifi:8.1.0/M1AJQ/T830XXU2ARJ3:user/release-keys'
2018-12-10 09:52:52.560 11276-11276/? A/DEBUG: Revision: '7'
2018-12-10 09:52:52.560 11276-11276/? A/DEBUG: ABI: 'arm'
2018-12-10 09:52:52.560 11276-11276/? A/DEBUG: pid: 9890, tid: 11225, name: ImageReader-640  >>> com.example.application <<<
2018-12-10 09:52:52.560 11276-11276/? A/DEBUG: signal 6 (SIGABRT), code -6 (SI_TKILL), fault addr --------
2018-12-10 09:52:52.565 11276-11276/? A/DEBUG: Abort message: 'statusor.cc:21 Attempting to fetch value instead of handling error generic::invalid_argument: CLOCK_MONOTONIC is ahead of CLOCK_BOOTTIME by 5.104us. Max is 5us'
2018-12-10 09:52:52.565 11276-11276/? A/DEBUG:     r0 00000000  r1 00002fcd  r2 00000006  r3 00000008
2018-12-10 09:52:52.565 11276-11276/? A/DEBUG:     r4 00001cc3  r5 00002fcd  r6 c09ff4a4  r7 0000010c
2018-12-10 09:52:52.565 11276-11276/? A/DEBUG:     r8 00000000  r9 cbb70480  sl c09ff8fc  fp 00000000
2018-12-10 09:52:52.565 11276-11276/? A/DEBUG:     ip c09ff5e8  sp c09ff490  lr f2ebb417  pc f2eb4abc  cpsr 20010030
2018-12-10 09:52:52.645 11276-11276/? A/DEBUG: backtrace:
2018-12-10 09:52:52.645 11276-11276/? A/DEBUG:     #00 pc 0001aabc  /system/lib/libc.so (abort+63)
2018-12-10 09:52:52.645 11276-11276/? A/DEBUG:     #01 pc 009953a1  /data/app/com.google.ar.core-RcvV0-pjyMCz3wIQPwjK5w==/lib/arm/libarcore_c.so
2018-12-10 09:52:52.645 11276-11276/? A/DEBUG:     #02 pc 009123e9  /data/app/com.google.ar.core-RcvV0-pjyMCz3wIQPwjK5w==/lib/arm/libarcore_c.so
2018-12-10 09:52:52.646 11276-11276/? A/DEBUG:     #03 pc 000bbc3f  /data/app/com.google.ar.core-RcvV0-pjyMCz3wIQPwjK5w==/lib/arm/libarcore_c.so
2018-12-10 09:52:52.646 11276-11276/? A/DEBUG:     #04 pc 000bc93b  /data/app/com.google.ar.core-RcvV0-pjyMCz3wIQPwjK5w==/lib/arm/libarcore_c.so
2018-12-10 09:52:52.646 11276-11276/? A/DEBUG:     #05 pc 000c010b  /data/app/com.google.ar.core-RcvV0-pjyMCz3wIQPwjK5w==/lib/arm/libarcore_c.so
2018-12-10 09:52:52.646 11276-11276/? A/DEBUG:     #06 pc 0000e069  /system/lib/libmediandk.so (AImageReader::CallbackHandler::onMessageReceived(android::sp<android::AMessage> const&)+156)
2018-12-10 09:52:52.646 11276-11276/? A/DEBUG:     #07 pc 00010649  /system/lib/libstagefright_foundation.so (android::AHandler::deliverMessage(android::sp<android::AMessage> const&)+24)
2018-12-10 09:52:52.646 11276-11276/? A/DEBUG:     #08 pc 00012b3d  /system/lib/libstagefright_foundation.so (android::AMessage::deliver()+60)
2018-12-10 09:52:52.646 11276-11276/? A/DEBUG:     #09 pc 000112af  /system/lib/libstagefright_foundation.so (android::ALooper::loop()+470)
2018-12-10 09:52:52.646 11276-11276/? A/DEBUG:     #10 pc 0000d343  /system/lib/libutils.so (android::Thread::_threadLoop(void*)+270)
2018-12-10 09:52:52.646 11276-11276/? A/DEBUG:     #11 pc 00075325  /system/lib/libandroid_runtime.so (android::AndroidRuntime::javaThreadShell(void*)+80)
2018-12-10 09:52:52.646 11276-11276/? A/DEBUG:     #12 pc 00048d33  /system/lib/libc.so (__pthread_start(void*)+22)
2018-12-10 09:52:52.646 11276-11276/? A/DEBUG:     #13 pc 0001b8df  /system/lib/libc.so (__start_thread+32)
```

**Comment by dev4711** [comment]
@AlgoRyan In case of the crash above I deployed with Sceneform 1.6.0. But I had the same crash also with Sceneform 1.5.1 and ARCore 1.5.3.
I will try to reproduce this crash also in an example project.

**Comment by dev4711** [comment]
@AlgoRyan unfortunatly I couldn't reproduce this crash on our app with Sceneform 1.6.0 or 1.5.1 and ARCore 1.6.181112096 anymore. With Sceneform 1.5.1 and ARCore 1.5.3 I was able to reproduce it. So this issue seams to be fixed already in 1.6. I also looked up the crash in the PlayStore and it didn't occur since last week (about the time ARCore 1.6 was released).
Either it was a one time crash on 1.6 or I mixed the versions on this crash and it was actually on 1.5. I'm sorry for this and will close this issue.
